diff --git BUILD.bazel BUILD.bazel
new file mode 100644
index 000000000..e7d27c141
--- /dev/null
+++ BUILD.bazel
@@ -0,0 +1,76 @@
+cc_library(
+    name = "common",
+    srcs = [
+        "lib/md5.c",
+        "lib/randutils.c",
+        "lib/sha1.c",
+        "lib/strutils.c",
+    ],
+    hdrs = [
+        "include/all-io.h",
+        "include/bitops.h",
+        "include/c.h",
+        "include/md5.h",
+        "include/nls.h",
+        "include/pathnames.h",
+        "include/randutils.h",
+        "include/sha1.h",
+        "include/strutils.h",
+    ],
+    defines = ["HAVE_NANOSLEEP"] +
+              select({
+                  "@platforms//os:linux": [
+                      "HAVE_SYS_FILE_H",
+                      "HAVE_REALLOCARRAY",
+                  ],
+                  "//conditions:default": [],
+              }),
+    strip_include_prefix = "include",
+    visibility = ["//visibility:public"],
+)
+
+cc_library(
+    name = "uuid",
+    srcs = [
+        "include/all-io.h",
+        "include/bitops.h",
+        "include/c.h",
+        "include/md5.h",
+        "include/nls.h",
+        "include/pathnames.h",
+        "include/randutils.h",
+        "include/sha1.h",
+        "include/strutils.h",
+        "lib/md5.c",
+        "lib/randutils.c",
+        "lib/sha1.c",
+        "lib/strutils.c",
+        "libuuid/src/clear.c",
+        "libuuid/src/compare.c",
+        "libuuid/src/copy.c",
+        "libuuid/src/gen_uuid.c",
+        "libuuid/src/isnull.c",
+        "libuuid/src/pack.c",
+        "libuuid/src/parse.c",
+        "libuuid/src/unpack.c",
+        "libuuid/src/unparse.c",
+        "libuuid/src/uuidP.h",
+        "libuuid/src/uuid_time.c",
+        "libuuid/src/uuidd.h",
+    ],
+    hdrs = [
+        "libuuid/src/uuid.h",
+    ],
+    include_prefix = "uuid",
+    strip_include_prefix = "libuuid/src",
+    visibility = ["//visibility:public"],
+    deps = [
+        ":common",
+    ],
+)
+
+alias(
+    name = "libuuid",
+    actual = ":uuid",
+    visibility = ["//visibility:public"],
+)
