diff --git a/repositories/rosbag2.BUILD.bazel b/repositories/rosbag2.BUILD.bazel
index 0a3db1b..a9c196c 100644
--- a/repositories/rosbag2.BUILD.bazel
+++ b/repositories/rosbag2.BUILD.bazel
@@ -15,15 +15,15 @@ load("@com_github_mvukov_rules_ros2//ros2:plugin.bzl", "ros2_plugin")
 load("@rules_python//python:defs.bzl", "py_library")
 load("@rules_ros2_pip_deps//:requirements.bzl", "requirement")
 
-ros2_cpp_library(
-    name = "rosbag2_storage",
+ros2_cpp_binary(
+    name = "rosbag2_storage.so",
     srcs = glob([
+        "rosbag2_storage/include/**/*.hpp",
         "rosbag2_storage/src/**/*.cpp",
         "rosbag2_storage/src/**/*.hpp",
     ]),
-    hdrs = glob(["rosbag2_storage/include/**/*.hpp"]),
     includes = ["rosbag2_storage/include"],
-    visibility = ["//visibility:public"],
+    linkshared = True,
     deps = [
         "@ros2_pluginlib//:pluginlib",
         "@ros2_rcpputils//:rcpputils",
@@ -32,6 +32,17 @@ ros2_cpp_library(
     ],
 )
 
+ros2_cpp_library(
+    name = "rosbag2_storage",
+    srcs = ["rosbag2_storage.so"],
+    hdrs = glob(["rosbag2_storage/include/**/*.hpp"]),
+    includes = ["rosbag2_storage/include"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":rosbag2_storage.so",
+    ],
+)
+
 ros2_plugin(
     name = "rosbag2_storage_default_plugins",
     srcs = glob([
