--- /dev/null
+++ BUILD.bazel
@@ -0,0 +1,26 @@
+load("@rules_cc//cc:defs.bzl", "cc_library", "cc_proto_library")
+load("@rules_proto//proto:defs.bzl", "proto_library")
+
+proto_library(
+    name = "protos",
+    srcs = glob(["foxglove/*.proto"]),
+    deps = [
+        "@com_google_protobuf//:duration_proto",
+        "@com_google_protobuf//:timestamp_proto",
+    ],
+)
+
+cc_proto_library(
+    name = "protos_cc",
+    deps = [":protos"],
+)
+
+# Group everything into a separate CC library rule so we can use the "includes"
+# attribute to enable the use of angular brackets when including the generated
+# headers
+cc_library(
+    name = "foxglove_schemas",
+    includes = ["."],
+    visibility = ["//visibility:public"],
+    deps = [":protos_cc"],
+)
