diff --git a/BUILD b/BUILD
new file mode 100644
index 0000000..e738bc5
--- /dev/null
+++ b/BUILD
@@ -0,0 +1,18 @@
+""" Builds pluginlib.
+"""
+
+load("@com_github_mvukov_rules_ros2//ros2:cc_defs.bzl", "ros2_cpp_library")
+
+ros2_cpp_library(
+    name = "pluginlib",
+    hdrs = glob(["pluginlib/include/**/*.hpp"]),
+    includes = ["pluginlib/include"],
+    visibility = ["//visibility:public"],
+    deps = [
+        "@ros2_ament_index//:ament_index_cpp",
+        "@ros2_class_loader//:class_loader",
+        "@ros2_rcpputils//:rcpputils",
+        "@ros2_rcutils//:rcutils",
+        "@tinyxml2",
+    ],
+)
